pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -6,35 +6,49 @@ import (
 	"crypto/x509"
 	"net/http"
 
+	"github.com/pkg/errors"
+
 	"github.com/coreos/go-oidc/v3/oidc"
 )
 
 func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.Context, error) {
-	if certificate != "" && key != "" {
-		certKeyClient := &http.Client{}
-		err := GetClientWithCertKey(certKeyClient, certificate, key)
-		if err != nil {
-			return nil, err
-		}
-		return oidc.ClientContext(ctx, certKeyClient), nil
+	certKeyClient, err := GetClientWithCertKey(certificate, key)
+	if err != nil {
+		return nil, err
+	}
+	return oidc.ClientContext(ctx, certKeyClient), nil
+}
+
+func GetClientWithCertKey(certificate, key string) (*http.Client, error) {
+	// Use the default transport to pick up on things like http.ProxyFromEnvironment
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	tlsConfig, err := getTLSClientConfig(certificate, key)
+	if err != nil {
+		return &http.Client{Transport: transport}, err
 	}
```

Comment:
Oh whoops just realized this won't work since the calling function returns nil if there's an error. Gonna fix this tonight.